src/main/java/io/github/syst3ms/skriptparser/util/math/NumberMath.java

```diff
@@ -200,6 +201,50 @@ public static Number tanh(Number n) {
         }
     }
 
+    /**
+     * Checks if a given BigInteger is prime. The certainty of this check is always 100%
+     * @param number the number to check
+     * @return whether or not this number is a prime
+     */
+    public static boolean isPrime(BigInteger number) {
+        // 1. Check if the number is of the form 6x±1 for values higher than 3.
+        if (!number.subtract(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && !number.add(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && number.compareTo(BigInteger.valueOf(3)) > 0)
+            return false;
+
+        // 2. Check in the cached primes.
+        if (cachedPrimes.contains(number.intValue()))
+            return true;
+
+        // 3. If the number is not too high, we can compute new primes and add them to the cached list.
+        // Prevents costly ln() invocation.
+        BigInteger threshold =
+                number.intValue() - cachedPrimes.getLast() < 100_000
```

Comment:
10,000 is a safer value, 100,000 is a bit on the expensive side.

Reply:
At 100K, the value (using your formula) was like 1.5M, which was surprisingly fast when I tested it.

I'll change it to 10K since my knowledge of this approaches zero, but I do have to say I did some testing and the value seemed alright to me.